README.md

```diff
@@ -143,25 +144,48 @@ let (simple, _full) =
 ## useMutation
 
 ```reason
-module ScreamMutationConfig = [%graphql {|
+module ScreamMutation = [%graphql {|
   mutation ScreamMutation($screamLevel: Int!) {
     scream(level: $screamLevel) {
       error
     }
   }
 |}];
 
-module ScreamMutation = ReasonApolloHooks.Mutation.Make(ScreamMutationConfig);
+[@react.component]
+let make = () => {
+  /* Both variant and records available */
+  let ( screamMutation, _simple, _full ) = useMutation(ScreamMutation.make(~screamLevel=10, ()), ());
+  let scream = (_) => {
+    screamMutation()
+      |> Js.Promise.then_(result => {
+          switch(result) {
+            | Data(data) => ...
+            | Error(error) => ...
+            | NoData => ...
+          }
+          Js.Promise.resolve()
+        })
+      |> ignore
+  }
 
+  <div>
+    <button onClick={scream}>
+      {React.string("You kids get off my lawn!")}
+    </button>
+  </div>
+}
+```
+
+If you don't know the value of the variables you can initialize with the query property
```

Comment:
there is no query property now

Reply:
> About useDynamicMutation - it returns a function without any query assign. For example you name the function: setUser, and pass to it something diffrent (for example adress) and it will work.
> Im just wondering if it will be good ?
> In react apollo you pass query to useMutation and it returns function which only takes variables, so here there is no possible to do mismatch which i wrote above.

Actually in apollo you can pass in a different mutation as options. I agree it's better to pass in the mutation once, and then only change variables, but they are conflated in `graphql_ppx`. I think for the longer term it's a good idea to suggest a PR to graphql_ppx to separate them (such as `Mutation.makeVariables(~id)` and `Mutation.mutation` (now `Mutation.query`).

Reply:
Maybe it's indeed better to future proof the API, to pass in `Mutation.query` as the first argument and a required named argument of variables of `Mutation.make(~id)##variables`. Then make that API prettier eventually by PR-ing `graphql_ppx`. What do you think?

Reply:
Another option is something similar what you made before: pass `Mutation.query` to useDynamicMutation and function which is returned can take `Mutation.make(~id)` (not only variables). But maybe its better to leave it as is and wait for better `graphql_ppx`.